examples/read_pem_files/main.rs

```diff
@@ -0,0 +1,63 @@
+use config::{Config, Format, Map, Value, ValueKind, FileStoredFormat};
+
+/// Global settings for the exposing all preconfigured variables
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct Settings {
+    pub private_key: Option<String>,
+    pub public_key: Option<String>,
+}
+
+fn get_settings() -> Result<Settings, config::ConfigError> {
+    let settings = Config::builder()
+        // read public key
+        .add_source(config::File::new("examples/read_pem_files/cert/public.pem", PemFile).required(false))
+        // read private key
+        .add_source(config::File::new("examples/read_pem_files/cert/private.pem", PemFile).required(false))
+        .build()
+        .unwrap();
+
+    settings
+        .try_deserialize::<Settings>()
+}
```

Comment:
- **`get_settings()`** into `main()`, as I think it makes sense to have `main()` presented early, and then follow the implementation further down below.
- Given the example is focused on demonstrating multiple sources to construct the config (_and that they're not both required_), I wanted to focus on that separately, so I've pulled them above the builder. With added comments, this breaks it down into 3 distinct steps: source -> build -> apply.
- **`try_deserialize()`** now with an `unwrap()` instead of `expect()`, only because you've not handled it earlier with `Config::builder()`. Could be `.expect()` but I don't think that's contributing much to the example.
- **`println!()`** now with `{:#?}` since `Debug` is derived, the output is slightly better presented this way (_not ideal due to the option, so we don't get proper multi-line rendering of the string content, line endings are rendered_).

```suggestion
fn main() {
    // Sourcing from two separate files for the `Settings` struct,
    // these are both marked as optional to avoid failure if the file doesn't exist.
    let file_public_key = config::File::new("public.pem", PemFile).required(false);
    let file_private_key = config::File::new("private.pem", PemFile).required(false);

    // Provide the sources and build the config object:
    let settings = Config::builder()
        .add_source(public_key)
        .add_source(private_key)
        .build()
        .unwrap();

    // Deserialize the config object into your Settings struct:
    let settings: Settings = settings.try_deserialize().unwrap();
    println!("{:#?}", settings);
}
```